### PR TITLE
feat: CP support Badge className / classNames and style / styles

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

`className` 作用于外面的 `<span />` ，`classNames.count` 作用于里面的 `<sup />`

`style` 本身就是挂在里面的 `<sup />` 上，`styles.count` 也作用于里面的 `<sup />`，所以和 `style` 和 `styles` 重合

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Badge className and style |
| 🇨🇳 Chinese | feat: CP 支持 Badge 组件的 className 和 style 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 9f01c70</samp>

Added global configuration for `Badge` component using `badge` prop in `ConfigProvider`. Updated `Badge` component to use `ConfigContext` for style and class name. Added test case and documentation for the new feature.

### <samp>🤖 Generated by Copilot at 9f01c70</samp>

*  Add `badge` property to `ConfigProvider` component and `ConfigContext` to allow users to set global configuration for `Badge` component ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR102), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R151), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R255), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R321))
*  Apply global configuration for `Badge` component in `style`, `className`, and `badge` attributes of relevant elements in `components/badge/index.tsx` ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L65-R65), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L116-R116), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R127), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R172))
*  Reorder import statements in `components/badge/index.tsx` to group type imports together and separate them from value imports ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L5-R10))
*  Adjust indentation of `getPrefixCls` function in `globalConfig` value in `components/config-provider/index.tsx` to match ESLint rule ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L203-R206))
*  Import `Badge` component in `components/config-provider/__tests__/style.test.tsx` and add test case to check that it respects global configuration ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R5), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R326-R336))
*  Document usage and default value of `badge` property in `Component Config` section in `components/config-provider/index.en-US.md` and `组件配置` section in `components/config-provider/index.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R105), [link](https://github.com/ant-design/ant-design/pull/43197/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R107))

### CODE REVIEW

File: components/badge/index.tsx [hunk 1]
```diff
@@ -124,6 +124,8 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
     }
 
     return {
+      ...badge?.style,
```
Comment (MadCcc) [inline components/badge/index.tsx:127]: 这个改了吧，changelog 注明一下 breaking change

Comment (MadCcc) [reply components/badge/index.tsx:127]: 新开个 PR 吧

File: components/badge/index.tsx [hunk 2]
```diff
@@ -123,10 +123,7 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
       offsetStyle.right = -parseInt(offset[0] as string, 10);
     }
 
-    return {
-      ...offsetStyle,
-      ...style,
-    };
+    return { ...badge?.style, ...offsetStyle, ...style };
```
Comment (MadCcc) [inline components/badge/index.tsx:126]: 少 depenency 了

File: components/badge/index.tsx [hunk 3]
```diff
@@ -168,6 +165,8 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
 
   const badgeClassName = classNames(
     prefixCls,
+    badge?.className,
+    badge?.classNames?.container,
```
Comment (MadCcc) [inline components/badge/index.tsx:169]:
```suggestion
    badge?.classNames?.root,
```

File: components/badge/index.tsx [hunk 4]
```diff
@@ -182,7 +181,11 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
   if (!children && hasStatus) {
     const statusTextColor = mergedStyle.color;
     return wrapSSR(
-      <span {...restProps} className={badgeClassName} style={mergedStyle}>
+      <span
+        {...restProps}
+        className={badgeClassName}
+        style={{ ...badge?.styles?.container, ...mergedStyle }}
```
Comment (MadCcc) [inline components/badge/index.tsx:187]:
```suggestion
        style={{ ...badge?.styles?.root, ...mergedStyle }}
```

File: components/badge/index.tsx [hunk 5]
```diff
@@ -194,7 +197,7 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
   }
 
   return wrapSSR(
-    <span ref={ref} {...restProps} className={badgeClassName}>
+    <span ref={ref} {...restProps} className={badgeClassName} style={badge?.styles?.container}>
```
Comment (MadCcc) [inline components/badge/index.tsx:200]: 这里少了 styles

File: components/config-provider/context.ts [hunk 6]
```diff
@@ -41,6 +41,17 @@ export interface ComponentStyleConfig {
...
 
+export interface BadgeConfig extends ComponentStyleConfig {
+  classNames?: {
+    count?: string;
+    container?: string;
+  };
+  styles?: {
+    count?: React.CSSProperties;
+    container?: React.CSSProperties;
+  };
```
Comment (MadCcc) [inline components/config-provider/context.ts:52]:
```suggestion
  classNames?: BadgeProps['classNames'];
  styles?: BadgeProps['styles'];
```

File: components/badge/index.tsx [hunk 7]
```diff
@@ -193,8 +195,15 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
   if (!children && hasStatus) {
     const statusTextColor = mergedStyle.color;
     return wrapSSR(
-      <span {...restProps} className={badgeClassName} style={{ ...styles?.root, ...mergedStyle }}>
-        <span className={statusCls} style={{ ...styles?.indicator, ...statusStyle }} />
+      <span
+        {...restProps}
+        className={badgeClassName}
+        style={{ ...styles?.root, ...badge?.style, ...badge?.styles?.root, ...mergedStyle }}
```
Comment (MadCcc) [inline components/badge/index.tsx:201]: `badge.style` 可以写到 mergedStyle 里

Comment (li-jia-nan) [reply components/badge/index.tsx]: done~

File: components/badge/index.tsx [hunk 8]
```diff
@@ -133,11 +133,8 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
       offsetStyle.right = -parseInt(offset[0] as string, 10);
     }
 
-    return {
-      ...offsetStyle,
-      ...style,
-    };
-  }, [direction, offset, style]);
+    return { ...badge?.style, ...offsetStyle, ...style };
```
Comment (MadCcc) [inline components/badge/index.tsx:136]: badge.style 应该在 offsetStyle 后面吧

Comment (li-jia-nan) [reply components/badge/index.tsx:136]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**